### ISSUE TITLE
Strange placement of one-digit index in boolean expression

### ISSUE THREAD

**Issue by costika1234** [psf/black#2947]
**Describe the style change**
I've just encountered an odd situation where `black` places a list index on a completely separately line. 

**Examples in the current _Black_ style**
Under default parameters, the following code snippet
```python
def f(special_chars):
    return (len([c for c in special_chars[1:] if c.isupper()]) > 0) and special_chars[0].islower()
```
is converted by `black 22.1.0` to:
```python
def f(special_chars):
    return (len([c for c in special_chars[1:] if c.isupper()]) > 0) and special_chars[
        0
    ].islower()
```

**Desired style**
The above is clearly not ideal as `0` is placed on its own line. I imagined that `black` would format the original function along the lines of:

```python
def f(special_chars):
    return (
        (len([c for c in special_chars[1:] if c.isupper()]) > 0)
        and special_chars[0].islower()
    )
```

**Additional context**
That being said, if the original snippet is changed to: 
```python
def f(special_chars):
    return len([c for c in special_chars[1:] if c.isupper()]) > 0 and special_chars[0].islower()
```
(that is, if we remove the parentheses around the inequality condition), then `black` formats everything correctly:
```python
def f(special_chars):
    return (
        len([c for c in special_chars[1:] if c.isupper()]) > 0
        and special_chars[0].islower()
    )
```
Is this a bug or not?

**Comment by FichteFoll** [comment]
Looks related to #1094.

**Comment by wbolster** [comment]
this particular example can also be rewritten so that it actually fits on one line 🙃

``` python
    return any(c.isupper() for c in special_chars[1:]) and special_chars[0].islower()
```

anyway, i believe this can be closed as a duplicate of #2156. if there are more than 2 expressions the expression will be split over multiple lines, e.g. with 3

``` python
    return (
        any(c.isupper() for c in special_chars[1:])
        and special_chars[0].islower()
        and True
    )
```

and 4 and so on:

``` python
    return (
        any(c.isupper() for c in special_chars[1:])
        and special_chars[0].islower()
        and True
        and True
    )
```

**Comment by felix-hilden** [comment]
Also related to #236. Feel free to close if you'd be happy with any of the proposed duplicate issues being resolved!
